File: tessoku/python/section3/b13.py
N, K = map(int, input().split())

As = [0] + list(map(int, input().split()))

cnt = 0

# 累積和を計算
cumsum = [0] * (N+1+1) # 0番目の要素を追加しておく

for i in range(1, N+1):
    cumsum[i] = cumsum[i-1] + As[i]

R = 1


for i in range(1,N+1):
    
    sum = cumsum[R] - cumsum[i-1]
    
    while R <= N and sum <= K:
        R += 1
        sum = cumsum[R] - cumsum[i-1]
    
    R -= 1 # sumがKを超えたらRを戻す，メモ：模範解答はRを戻す代わりに，cumsum[R+1] - cumsum[i-1]としている
    
    cnt += R - i + 1
    
print(cnt)

File: tessoku/c++/section3/b13.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {

    int N;
    long long K;

    cin >> N >> K;

    long long As[100009];

    for (int i = 1; i <= N; i++) {
        cin >> As[i];
    }

    long long cnt = 0;

    long long R = 0; // 初期値は0

    long long sum = 0;

    
    for (int i = 1; i <= N; i++) {

        while (sum <= K){
            R++;
            sum += As[R];
            
            // 最大データ数を超えたらbreak
            if (R > N) {
                break;
            }

        }


        // sumを戻す
        sum -= As[R];
        sum -= As[i];

        // Kを超えたらRを戻す
        R--;
        
        cnt += R - i + 1; // 連続とあるが，1つだけでもOKらしい

    }

    cout << cnt << endl;
    
}

